### PR TITLE
fix: send confirmation email on re-adding agent

### LINKED ISSUE #4951: Re-adding an agent doesn't trigger activation email
**Describe the bug**

If an agent got deleted and added again, he is not receiving an activation email.

**To Reproduce**

1. add agent
2. delete agent
3. add agent again

**Expected behavior**

Agent should receive an activation email

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Send confirmation email after re-adding the agent.

Fixes #4951 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Tested on local
- Added test cases for the same

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
https://www.loom.com/share/1c363a3be97944f8a2fdcc8dd2a6b493

**Comment by tejaswinichile** [comment]
https://www.loom.com/share/4f97d1e0a0f64143a11fae8c43cc727c

Send `login link` to the user when the account has already been verified before.

File: spec/mailers/confirmation_instructions_spec.rb [hunk 1]
```diff
@@ -61,5 +62,17 @@
         expect(confirmable_user.unconfirmed_email.blank?).to be false
       end
     end
+
+    context 'when user deleted before' do
```
Comment (sojan-official) [inline spec/mailers/confirmation_instructions_spec.rb:66]: `when the user is already confirmed` ?

File: spec/mailers/confirmation_instructions_spec.rb [hunk 2]
```diff
@@ -61,5 +62,17 @@
...
+      before do
+        confirmable_user.confirm
+        confirmable_user.account_users.last.destroy!
+      end
+
+      it 'send login after re-adding confirmed agent' do
```
Comment (pranavrajs) [inline spec/mailers/confirmation_instructions_spec.rb:72]:
```suggestion
      it 'send instructions with login link' do
```

**Review by sojan-official — APPROVED** [review]

**Comment by pranavrajs** [comment]
@tejaswinichile Can't this be merged? Is there anything pending here?

**Pull request merged**